src/memory.rs

```diff
@@ -127,7 +127,102 @@ pub fn to_java_heap_value(
     }
 }
 
+/// Parsed representation of a K8s memory/storage resource limit.
+#[derive(Clone, Copy, Debug)]
+pub struct MemoryQuantity {
+    pub value: f32,
+    pub unit: BinaryMultiple,
+}
+
 impl MemoryQuantity {
+    pub fn from_gibi(gibi: f32) -> Self {
+        Self {
+            value: gibi,
+            unit: BinaryMultiple::Gibi,
+        }
+    }
+
+    pub fn from_mebi(mebi: f32) -> Self {
+        Self {
+            value: mebi,
+            unit: BinaryMultiple::Mebi,
+        }
+    }
+
+    /// Scales down the unit to GB if it is TB or bigger.
+    /// Leaves the quantity unchanged otherwise.
+    fn scale_to_at_most_gb(&self) -> Self {
+        match self.unit {
+            BinaryMultiple::Kibi => *self,
+            BinaryMultiple::Mebi => *self,
+            BinaryMultiple::Gibi => *self,
+            BinaryMultiple::Tebi => self.scale_to(BinaryMultiple::Gibi),
+            BinaryMultiple::Pebi => self.scale_to(BinaryMultiple::Gibi),
+            BinaryMultiple::Exbi => self.scale_to(BinaryMultiple::Gibi),
+        }
+    }
+
+    /// Scale down the unit by one order of magnitude, i.e. GB to MB.
+    fn scale_down_unit(&self) -> OperatorResult<Self> {
+        match self.unit {
+            BinaryMultiple::Kibi => Err(Error::CannotScaleDownMemoryUnit),
+            BinaryMultiple::Mebi => Ok(self.scale_to(BinaryMultiple::Kibi)),
+            BinaryMultiple::Gibi => Ok(self.scale_to(BinaryMultiple::Mebi)),
+            BinaryMultiple::Tebi => Ok(self.scale_to(BinaryMultiple::Gibi)),
+            BinaryMultiple::Pebi => Ok(self.scale_to(BinaryMultiple::Tebi)),
+            BinaryMultiple::Exbi => Ok(self.scale_to(BinaryMultiple::Pebi)),
+        }
+    }
+
+    /// Floors the value of this MemoryQuantity.
+    pub fn floor(&self) -> Self {
+        Self {
+            value: self.value.floor(),
+            unit: self.unit,
+        }
+    }
+
+    /// If the MemoryQuantity value is smaller than 1 (starts with a zero), convert it to a smaller
+    /// unit until the non fractional part of the value is not zero anymore.
+    /// This can fail if the quantity is smaller than 1kB.
+    fn ensure_no_zero(&self) -> OperatorResult<Self> {
+        if self.value < 1. {
+            self.scale_down_unit()?.ensure_no_zero()
+        } else {
+            Ok(*self)
+        }
+    }
```

Comment:
The value could be 0 here? If yes this will loop forever. I would add a check for this.